src/main/java/com/googlecode/download/maven/plugin/internal/WGet.java

```diff
@@ -300,46 +298,23 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         this.outputDirectory.mkdirs();
         File outputFile = new File(this.outputDirectory, this.outputFileName);
 
+        final Signatures signatures = new Signatures(
+            this.md5, this.sha1, this.sha256, this.sha512, this.getLog()
+        );
         // DO
         try {
             boolean haveFile = outputFile.exists();
             if (haveFile) {
                 boolean signatureMatch = true;
                 if (this.checkSignature) {
-                    String expectedDigest = null, algorithm = null;
-                    if (this.md5 != null) {
-                        expectedDigest = this.md5;
-                        algorithm = "MD5";
-                    }
-
-                    if (this.sha1 != null) {
-                        expectedDigest = this.sha1;
-                        algorithm = "SHA1";
-                    }
-
-                    if (this.sha256 != null) {
-                        expectedDigest = this.sha256;
-                        algorithm = "SHA-256";
-                    }
-
-                    if (this.sha512 != null) {
-                        expectedDigest = this.sha512;
-                        algorithm = "SHA-512";
-                    }
-
-                    if (expectedDigest != null) {
-                        try {
-                            SignatureUtils.verifySignature(outputFile, expectedDigest,
-                                MessageDigest.getInstance(algorithm));
-                        } catch (MojoFailureException e) {
-                            getLog().warn("The local version of file " + outputFile.getName() + " doesn't match the expected signature. " +
-                                "You should consider checking the specified signature is correctly set.");
-                            signatureMatch = false;
-                        }
+                    try {
+                        signatures.validate(outputFile);
```

Comment:
And if we have now only one usage of `SignatureUtils.verifySignature` then we can modify it and return Optional<String> with errors instead of throwing an exception. It is cheaper.

Reply:
@vik4es exception handling here is rather messy. I'd address the concern in a separate issue